### PR TITLE
Update for rust 1.10.0-nightly

### PR DESCRIPTION
**Pull request by agalakhov**
This fixes platformtree!, ioreg! and lang items.

### CODE REVIEW

File: src/util/lang_items.rs [hunk 1]
```diff
@@ -17,18 +17,10 @@
 use core::fmt::Arguments;
 
 #[cfg(all(not(test), not(feature = "test")))]
-#[lang="stack_exhausted"]
-extern fn stack_exhausted() {}
```
Comment (farcaller) [inline src/util/lang_items.rs]: is this lang item just gone now?

**Comment by farcaller** [comment]
wow, llvm assert, haven't seen those in a while.

File: src/util/lang_items.rs (hunk 1 above)
Comment (agalakhov) [inline src/util/lang_items.rs]: Yes, removed from rust by commit 7a3fdfbf "remove morestack support"

Comment (farcaller) [inline src/util/lang_items.rs]: Thanks for the reference.

**Comment by farcaller** [comment]
A great work overall, much appreciated. I'm tempted to wait a day until a new nightly to see if a fresh build will solve the llvm issue and report it upstream otherwise.

**Comment by agalakhov** [comment]
Thank you. I believe the LLVM issue could be solved by editing the data-layout in json. Looks like that this line is not needed anymore (but I'm not really sure).

**Comment by jmagnuson** [comment]
`data-layout` recently became mandatory again, as discussed in rust-lang/rust#31367.  I was able to successfully compile @agalakhov's branch using the layout seen [here](https://github.com/thejpster/bare-metal-arm-rust/commit/e75add5d169d82797305c268ce1e215a845ab749) (for thumbv7em anyway).

**Comment by agalakhov** [comment]
Made a fix against LLVM assets. Not perfect, but now it at least works.

**Comment by farcaller** [comment]
Thanks for your contributions!

**Pull request merged**